src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+  }
+}).buildAllPlatforms();
+```
+
+This will first output a **variables.css** file with all design tokens and use `outputReferences` so that we only need to override a subset of CSS custom properties for dark mode to work. Then for dark mode it will output **variables-dark.css** with only tokens defined in files with `.dark` in the file path. In our web demo we can include **variables.css** by default and **variables-dark.css** when `prefers-color-scheme` is "dark"
+
+```html
+<link rel="stylesheet" href="variables.css">
+<link rel="stylesheet" href="variables-dark.css" media="(prefers-color-scheme: dark)">
+```
+
+If you wanted to control the color mode with a setting on your web app you could similarly output 2 files and use different selectors for the `css/variables` format:
+
+```javascript
+// dark mode
+StyleDictionary.extend({
+  // ... skipping configuration above
+  platforms: {
+    css: {
+      transformGroup: `css`,
+      buildPath: webPath,
+      files: [{
+        destination: `variables-dark.css`,
+        format: `css/variables`,
+        filter: (token) => token.filePath.indexOf(`.dark`) > -1,
+        options: {
+          outputReferences: true,
+          selector: '.dark'
+        }
+      }]
+    }
+  }
+})
+```
+
+We will also generate 2 sets of SVG graphics, one for light and one for dark mode. To do that we will use the same custom action, but add some configuration to let the action know if it is dark or light mode and write the SVG to different files.
+
+
+
+
+### Android
+
+This approach works well with Android because Android resources use different folders (values and values-night) for light/default and dark mode. This works the same way as our CSS files: we use `outputReferences` and for dark mode only output the subset of color tokens that have dark values. We will generate 2 color resource files:
+
+1. `values/colors.xml`
+1. `values-night/colors.xml`
+
+Here is the configuration to do that:
+
+```javascript
+// light/default mode
+StyleDictionary.extend({
+  //...
+  platforms: {
+    // ...
+    android: {
+      transformGroup: `android`,
+      buildPath: androidPath,
+      files: [{
+        destination: `values/colors.xml`,
+        format: `android/resources`,
+        filter: (token) => token.attributes.category === `color`,
+        options: {
+          outputReferences: true
+        },
+      }]
+    }
+  }
+}).buildAllPlatforms();
+ 
+// dark mode
+StyleDictionary.extend({
+  // ...
+  platforms: {
+    // ...
+    android: {
+      transformGroup: `android`,
+      buildPath: androidPath,
+      files: [{
+        destination: `values-night/colors.xml`,
+        format: `android/resources`,
+        // only outputting the tokens from files with '.dark' in the filepath
+        filter: (token) => token.filePath.indexOf(`.dark`) > -1
+      }]
+    }
+  }
+}).buildAllPlatforms();
+```
+
+Then using the same `generateGraphics` custom action we will generate 2 vector graphic resources per graphic:
```

Comment:
custom action we will generate 2 vector > custom action, we will generate two vector